apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -621,6 +621,34 @@ defmodule Champions.Battle.Simulator do
     [caster.id]
   end
 
+  defp choose_targets_by_strategy(
+         caster,
+         %{type: %{"lowest" => stat}, target_allies: target_allies, count: count},
+         state
+       ) do
+    target_team =
+      Enum.filter(state.units, fn {_id, unit} -> unit.team == caster.team == target_allies end)
+
+    Enum.map(target_team, fn {_id, unit} -> unit end)
+    |> sort_units_by_stat(stat, true)
+    |> Enum.take(count)
+    |> Enum.map(fn unit -> unit.id end)
+  end
+
+  defp choose_targets_by_strategy(
+         caster,
+         %{type: %{"highest" => stat}, target_allies: target_allies, count: count},
+         state
+       ) do
+    target_team =
+      Enum.filter(state.units, fn {_id, unit} -> unit.team == caster.team == target_allies end)
+
+    Enum.map(target_team, fn {_id, unit} -> unit end)
+    |> sort_units_by_stat(stat, false)
+    |> Enum.take(count)
+    |> Enum.map(fn unit -> unit.id end)
```

Comment:
This whole behavior is the same as above, it just changes the sort order. Can we wrap this up in a function, so each parent function calls it with the desired order to sort?